src/arrayMethodJoin.js

```diff
@@ -3,9 +3,36 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let sep = separator;
+
+    if (separator === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] === undefined) {
+        return str;
+      }
+
+      if (i === this.length - 1) {
+        str += this[i];
+
+        return str;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        str += ',';
+        continue;
+      }
+
+      str += this[i] + sep;
```

Comment:
you can write not so many conditions. Do like this 
```suggestion
      if (this[i] !== undefined && this[i] !== null) {
        str += this[i];
      }

      if (i < this.length - 1) {
        str += separator;
      }
```